libs/x-content/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -26,15 +24,16 @@
 public class ParseField {
     private final String name;
     private final String[] deprecatedNames;
-    private final Set<RestApiVersion> restApiVersions = new HashSet<>(2);
+    private final Function<RestApiVersion, Boolean> restApiVersionMatcher;
```

Comment:
any suggestions for a better name? restApiVersionMatcher? restApiVersionFunction? restApiVersionRange ?

Reply:
how about `forRestApiVersion` see comment below.